spaceship/spec/client_spec.rb

```diff
@@ -100,4 +112,11 @@ def subject.send_login_request(user, password)
       end
     end
   end
+
+  describe "#log_response" do
+    it 'handles ASCII-8BIT to UTF-8 encoding gracefully' do
+      response = TestResponse.new([130, 5, 3120, 130, 4, 171, 160, 3, 2].pack('C*'))
+      expect(subject.try_log_response(:get, TestClient.hostname, response)).to be_truthy
```

Comment:
The use of `private` in `spaceship` is somewhat unique in fastlane, most other classes don't use it to restrict access to internal methods. As such, there aren't really well established conventions for how to handle this particular case.

I think in this case, you might consider using `send` to achieve the same behavior. This would allow you not to have to use `try_log_response` and instead just call:
```expect(subject.send(:try_log_response, :get, TestClient.hostname, response)).to be_truthy```
Maybe throw a comment on the line of code explaining that you are using `send` because the `log_response` method is marked as private.